### ISSUE TITLE
hardware issue and solution

### ISSUE THREAD

**Issue by pepoulabarbe** [bennyblanc0/ebike_speed_sensor_interceptor#2]
Thanks for your good job!
So i had one issue with the sensor reading with a mini pro board and i make some modification/simplification that work perfectly now on my shimano ebike:
- Connect the sensor directly to the 5V pinout board (in place of the pin 11).
- Connect a 10k resistor between the sensor pin 12 and the ground (prevent erratic mode of the pin).
- Remove all the pin 11 correspondence in the code.

King regards,
Jean-Luc

**Comment by Mguss** [comment]
Are you using original sensor from Shimano? If so. You say that you put sensor om eleven (Red or Black)
If you connect like this you don’t need to take 5V from engine ?

Please Micke

**Comment by pepoulabarbe** [comment]
yes, it's the original sensor. We need to take the 6V from engine to powered the arduino board (raw arduino pinout). The shimano sensor is just a passive switch (like ILS) open and close by the magnet, the regulated 5v comming from the arduino board cross the sensor and come back to the pin 11 for count interrupt.

**Comment by Mguss** [comment]
Many thanks. Another question. Only two cables are used on the trim sold in the store. Do you know how the chip gets power?

**Comment by Mguss** [comment]
Sending over my code that’s works if someone want to use.

//----------------------------------------------------------------------------
  /*
    2019-04-29
    NO-LIMIT V1.4
    SPEED BARRIER UNLOCKER FOR SHIMANO E6000
    
    READ SIGNAL FROM HALL-SENSOR BACK WHEEL ON DIGITAL PIN 4
    SEND EVERY SIGNAL TO ECM ON DIGITAL PIN 8. IF SPEED IS OVER X (Variable SpeedChange)  KM/H SEND TO DIGITAL 8 EVERY OTHER (HALF SPEED)
  
    SETTING: Measure weel circumference on bike then set measure in variable "WheelSize" (ex. 2.23) for 2,23 meters
    help for circumference http://www.bikecalc.com/wheel_size_math
  */
//----------------------------------------------------------------------------
  #include <Bounce2.h>
  #define SENSOR_PIN 4
  #define MOTOR_PIN 8

  const String Version = "NO-LIMIT Ver: 1.4 by Guss";
  const float WheelSize = 2.133;                        	// "SETTING" Setting circumference for wheel 2,133 = 28"
  const int SpeedChange = 22;                           	// "SETTING" Max original speed, then we send everyOhter to half speed on odometer
  float kph = 0.00;                                     		// Calculated speed from SENSOR_PIN (kph = 3600000 / one wheel revolution time) * WheelSize / 1000)
  int kphRound = 0.00;
  float last_kph = 0.00;
  //float Top25_kph = 0.0;                                	// Get speed + X% (check for Bounce UPP from last kph)
  //float Low25_kph = 0.0;                                	// Get speed - X% (check for Bounce DOWN from last kph)
  int EveryOther = 1;
  unsigned long endTime=0;    
  unsigned long StartMillis;
  unsigned long nowMillis;
  Bounce debouncer = Bounce();                          	// Instantiate a Bounce object
//----------------------------------------------------------------------------
  void setup() {
    Serial.begin(2000000);
    debouncer.attach(SENSOR_PIN,INPUT_PULLUP);          // Attach the debouncer to a pin with INPUT_PULLUP mode
    pinMode(MOTOR_PIN, OUTPUT);                         		// OUTPUT pin to motor
    debouncer.interval(1);                              				// Use a debounce interval of X milliseconds (Test for best result)
    while (!Serial) {
      ;                                                 					// Wait for serial port to connect. Needed for native USB
    }
    
    Serial.println(Version);
  }
//----------------------------------------------------------------------------
  void loop() {
    debouncer.update();                                  				// Update the Bounce instance
    if ( debouncer.fell() ) {                            				// Call code if SENSOR_PIN transitions from HIGH to LOW
      
      nowMillis = millis();                             				 // Get time now (end time loop)           
      endTime = nowMillis - StartMillis;                 			// Time between every loop
      StartMillis = millis();                            				// Start time for loop (start time loop)
      kph = (3600000 / endTime) * WheelSize / 1000;      	// Calc speed in kph from Wheel revulotion
      kphRound=kph;                                      				// Make kph to integer

      if(kphRound <= SpeedChange){                       		// If speed i lower than SpeedChange Send every time                              
        SendOutIt();                                     				// Send to MOTOR
      }
      else {
        if(EveryOther > 1){                              				// If speed i higher than SpeedChange send every other to MOTOR
          SendOutIt();                                  				 // Send to MOTOR
          EveryOther = 0;                                				// Set EveryOther to 0
        }
        EveryOther++;                                   				 // EveryOther ++
      }
      Serial.println(kphRound);
    }
  }
//----------------------------------------------------------------------------
//SENDING TO MOTOR
  void SendOutIt(){
    digitalWrite(MOTOR_PIN, LOW);
    delay(1);                                           					// Try best value.
    digitalWrite(MOTOR_PIN, HIGH);
    Serial.println("motor");
  }

**Comment by Mguss** [comment]
Hi again. Is it possible to send me how you connect sensor ant the other cables on your mini pro?
Please Micke

**Comment by Mguss** [comment]
Hi everyone.
Great job. Is it possible to post a picture of the connection on than board?

Pleas Micke

**Comment by pepoulabarbe** [comment]
![schema cablage](https://user-images.githubusercontent.com/50272399/58506470-b21c0d00-818f-11e9-9aef-0db8471f74b3.jpg)

regards,

**Comment by hippylover** [comment]
Anyone got this working? Seems like people are using different solutions. Anyone have gotten this to run without getting these e014 error code after a while? Not sure i understand why you turn the motor off or on with a relay from arduino? (i thought this was just something that went inbetween the speed sensor and the input to the engine circuitry...)
